borsh/src/schema.rs

```diff
@@ -49,8 +49,25 @@ pub type FieldName = String;
 pub enum Definition {
     /// A fixed-size array with the length known at the compile time and the same-type elements.
     Array { length: u32, elements: Declaration },
-    /// A sequence of elements of length known at the run time and the same-type elements.
-    Sequence { elements: Declaration },
+
+    /// A dynamically-sized sequence of homogeneous elements.
+    ///
+    /// The sequence is encoded with a 4-byte length followed by the elements of
+    /// the sequence.  Prototypical example of type which uses this definition
+    /// is `Vec<T>`.
+    Sequence {
+        /// Bounds on the possible lengths of the sequence.
+        ///
+        /// For unbounded sequences (such as `Vec<T>`) it’s `0..=u32::MAX`
+        /// however custom types may have limits which can be specified as
+        /// minimum and maximum length.  For example, a `BridgePlayers` type may
+        /// be defined as a sequence of players with length range `3..=4`.
+        length_range: core::ops::RangeInclusive<u32>,
+
```

Comment:
i've skimmed through this. 
looks good.
this pr looks compact enough to also include change in the same one

> I’m gonig to add length_width and replace Array in next PR.

requesting to include this change too, as it will be easier to fix conflicts with https://github.com/near/borsh-rs/pull/219 only once, and not 2 times